app/src/main/java/com/kylecombes/micrepeater/MainActivity2.java

```diff
@@ -111,10 +111,20 @@ public void stateChanged(boolean deviceConnected, boolean scoAudioConnected,
     public void startAmplification() {
         Log.d(TAG, "Starting amplification...");
         audioRelayServiceIntent = new Intent(this, AudioRelayService.class);
-        Integer streamType = Objects.requireNonNull(mViewModel.getStreamType().getValue());
-        audioRelayServiceIntent.putExtra(AudioRelayService.STREAM_KEY, streamType);
-        // Set default volume control
-        setVolumeControlStream(streamType);
+        String streamType = getSharedPreferences("PREFERENCE", MODE_PRIVATE)
+                .getString("audioOutput", "voice");
+        if(streamType.equals("voice")){
+            audioRelayServiceIntent.putExtra(AudioRelayService.STREAM_KEY, AudioManager.STREAM_VOICE_CALL);
+            setVolumeControlStream(AudioManager.STREAM_VOICE_CALL);
+        } else if(streamType.equals("alarm")){
+            audioRelayServiceIntent.putExtra(AudioRelayService.STREAM_KEY, AudioManager.STREAM_ALARM);
+            setVolumeControlStream(AudioManager.STREAM_ALARM);
+        } else {
+            audioRelayServiceIntent.putExtra(AudioRelayService.STREAM_KEY, AudioManager.STREAM_MUSIC);
+            setVolumeControlStream(AudioManager.STREAM_MUSIC);
+        }
+//        Integer streamType = Objects.requireNonNull(mViewModel.getStreamType().getValue());
```

Comment:
Just remove this

Reply:
Also, go to `AppStateViewModel` class and remove anything related to `streamType`. We don't need it there anymore.

Reply:
wait why do we remove that...?